lib/async_hooks.js

```diff
@@ -174,15 +185,27 @@ class AsyncResource {
   }
 
   emitBefore() {
+    showEmitBeforeAfterWarning();
     emitBefore(this[async_id_symbol], this[trigger_async_id_symbol]);
     return this;
   }
 
   emitAfter() {
+    showEmitBeforeAfterWarning();
     emitAfter(this[async_id_symbol]);
     return this;
   }
 
+  makeCallback(fn, thisArg, ...args) {
+    emitBefore(this[async_id_symbol], this[trigger_async_id_symbol]);
+    try {
+      Reflect.apply(fn, thisArg, args);
+    } finally {
+      emitAfter(this[async_id_symbol]);
+    }
+    return this;
+  }
```

Comment:
Is there a reason to return `this` here rather than returning the return value of the function given to the `Reflect.apply(...)` call?

Reply:
That is good feedback, thanks! I was following the fluent convention that the rest of the class uses, but you're correct that it is better to return value produced by the callback.

Reply:
Done.